server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1908,7 +1908,8 @@ public void flush(boolean force, boolean waitIfOngoing) throws EngineException {
             if (flushLock.tryLock() == false) {
                 // if we can't get the lock right away we block if needed otherwise barf
                 if (waitIfOngoing == false) {
-                    return;
+                    logger.trace("returning as there is an in-flight flush that we do not need to wait for");
```

Comment:
nit: I'd like the log message to be less about the "return" statement and more about the situation:

```suggestion
                    logger.trace("detected an in-flight flush, not blocking to wait for it's completion");
```